react-components/tests/tests-utilities/fixtures/queryResult.ts

```diff
@@ -0,0 +1,22 @@
+import { type UseQueryResult } from '@tanstack/react-query';
+import { vi } from 'vitest';
+
+export const createMockQueryResult = <T>(data: T, isFetching = false): UseQueryResult<T> =>
+  ({
+    data,
+    isFetching,
+    isLoading: false,
+    isError: false,
+    error: null,
+    refetch: vi.fn()
+  }) as unknown as UseQueryResult<T>;
```

Comment:
I would prefer to e.g. use moq.ts to do this in a type safe way instead of casting to unknown